facia-press/app/frontpress/FapiFrontPress.scala

```diff
@@ -489,13 +453,62 @@ trait FapiFrontPress extends EmailFrontPress with Logging {
 
     contentApiResponse.map(Option(_)).fallbackTo(Future.successful(None))
   }
-
 }
 
-object CapiUrl {
-  def extractId(urlString: String): String = {
-    val uri = new URI(urlString)
-    val path = uri.getPath()
-    path.stripPrefix("/")
+object Enrichment extends Logging {
+  def enrichSnap(embedUri: Option[String], beforeEnrichment: EnrichedContent, collection: Collection, wsClient: WSClient)
+                (implicit executionContext: ExecutionContext): Future[EnrichedContent] = {
+    embedUri match {
```

Comment:
Not introduced here I realise, and not a blocker, but might be worth introducing some variables to help with readabiilty. E.g. something like:

```    embedUri match {
      case Some(embedUri) =>
        val response = wsClient.url(embedUri).get()
          
        val enriched = response.map { response =>
          Json.fromJson[EmbedJsonHtml](response.json) match {
            case JsSuccess(embed, _) =>
              beforeEnrichment.copy(embedHtml = Some(embed.html))
            case _ =>
              log.warn(s"An embed had invalid json format, and won't be pressed. $embedUri for collection ${collection.id}")
              beforeEnrichment
          }
        } 
        
        enriched.recover {
          case _ => {
            log.warn(s"A request to an embed uri failed, embed won't be pressed. $embedUri for collection ${collection.id}")
            beforeEnrichment
          }
        }
      case None => Future.successful(beforeEnrichment)
    }
```

In addition to new variables, I've also changed `case _` to `case None` as it can be nice to be explicit where possible and avoid `_` - as it helps the reader and also the compiler will warn if you've got it wrong usually.